### PR TITLE
CI: Fail on warning for 3.9

### PR DESCRIPTION
**Pull request by TomAugspurger**
This should fail till https://github.com/pandas-dev/pandas/pull/34835 is in.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ matrix:
     - dist: bionic
       python: 3.9-dev
       env:
-        - JOB="3.9-dev" PATTERN="(not slow and not network and not clipboard)"
+        - JOB="3.9-dev" PATTERN="(not slow and not network and not clipboard)" TEST_ARGS="-Werror"
```
Comment (WillAyd) [inline .travis.yml:36]: This gets passed to pytest right? I think the flag there is just `-W`

https://docs.pytest.org/en/latest/warnings.html

Comment (WillAyd) [reply .travis.yml:36]: Or could even do in the setup.cfg

Comment (TomAugspurger) [reply .travis.yml]: Trying it out in setup.cfg. The primary issue I see is that we don't necessarily want warnings from optional dependencies to cause errors. For now I've limited to just our required dependencies.

**Comment by WillAyd** [comment]
Just out of curiosity what are you trying to do specific to 3.9? I think great to do this, though obviously would require a good deal of work to get it to work across all builds

**Comment by TomAugspurger** [comment]
> Just out of curiosity what are you trying to do specific to 3.9?

The 3.9 job builds NumPy from source, and so it can be slightly ahead of the numpydev build (our only other build to error on warnings) which uses NumPy's nightly wheels. 

Agreed this would be nice to have. Just not sure if /when I'll have time to get to work through all of these. There are more failures than I expected :)

**Comment by TomAugspurger** [comment]
I'm not going to get to this any time soon.

**Pull request closed without merging**